### ISSUE TITLE
[User] ChatML Support

### ISSUE THREAD

**Issue by zz2115** [ggerganov/llama.cpp#3598]
Is it on the roadmap to support ChatML proposed by [openai](https://github.com/openai/openai-python/blob/main/chatml.md)?

**Comment by staviq** [comment]
Not officially ( I don't think so ), but I was planning on giving it a go after #3538 because it enables proper control/special token usage in prompt "templates".

Some kind of very simple per model prompt template / library system would be nice to have, I just need to figure out a way to do it cleanly and non-invasively.
